### PR TITLE
impl Hash for MemoryLocation and impl Sync for SubAllocation

### PR DESCRIPTION
**Pull request by termhn**
Hash is nice for using MemoryLocation to key hashmaps and `Sync` for `SubAllocation` should be safe for the same reason it is for `MemoryBlock`

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -197,6 +197,7 @@ pub struct SubAllocation {
 }
 
 unsafe impl Send for SubAllocation {}
+unsafe impl Sync for SubAllocation {}
```
[the remark this answers is not in the document]
Comment (Jasper-Bekkers) [reply src/lib.rs:200]: Sounds like a good idea @MarijnS95

[the remark this answers is not in the document]
Comment (termhn) [reply src/lib.rs:200]: yep :) I'll copy it over and push a commit, sec..

[the remark this answers is not in the document]
Comment (MarijnS95) [reply src/lib.rs:202]: Sorry, I did not mean "the same" verbatim :)

`mapped_ptr` _is_ exposed publicly by `SubAllocation` - it's just marked `unsafe` :smile:

[the remark this answers is not in the document]
Comment (MarijnS95) [reply src/lib.rs:205]: Thanks! Nitpick: maybe add backticks consistently around `mapped_ptr`, but it's a regular Rust comment and not a doc-comment so I don't _reaaally_ mind :grimacing:

**Pull request merged**